scipy/interpolate/_rbfinterp.py

```diff
@@ -407,17 +465,19 @@ def __call__(self, x):
 
         nx, ndim = x.shape
         if ndim != self.y.shape[1]:
-            raise ValueError(
-                "Expected the second axis of `x` to have length "
-                f"{self.y.shape[1]}."
-                )
-
+            raise ValueError("Expected the second axis of `x` to have length "
+                             f"{self.y.shape[1]}.")
+        # how many floats in memory we already occupy
+        # We use that as a limit for chunking the evaluation
+        memory_budget = x.size + self.y.size + self.d.size
         if self.neighbors is None:
-            out = _evaluate(
-                x, self.y, self.kernel, self.epsilon, self.powers, self._shift,
-                self._scale, self._coeffs
-                )
-
+            out = self._chunk_evaluator(
```

Comment:
It is a bit awkward and confusing that `_chunk_evaluator` is a method, but you still have to pass attributes in as arguments. My preference would be to have `_chunk_evaluator` as a function just like what `_evaluate` used to be.

Reply:
The are several common class attributes that are used inside chunk_evaluator, so I think the current interface make sense to me.